src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import static java.util.Objects.hash;
+
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(key, value, hash(key), null);
+        int bucketIndex = calculateIndex(newNode.key);
+        if (table[bucketIndex] == null) {
+            add(newNode, bucketIndex);
+        } else {
+            putNew(newNode, bucketIndex);
+        }
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int bucketIndex = calculateIndex(key);
+        Node<K, V> thisNode = table[bucketIndex];
+        while (thisNode != null) {
+            if (Objects.equals(thisNode.key, key)) {
+                return thisNode.value;
+            }
+            thisNode = thisNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void add(Node<K, V> node, int bucketIndex) {
```

Comment:
less confusing naming
```suggestion
    private void putByIndex(Node<K, V> node, int bucketIndex) {
```

Reply:
change the confusing naming -  from add to putByIndex